llama-index-integrations/vector_stores/llama-index-vector-stores-hologres/llama_index/vector_stores/hologres/base.py

```diff
@@ -0,0 +1,197 @@
+            connection_string=connection_string,
+            table_name=table_name,
+            embedding_dimension=embedding_dimension,
+            table_schema=table_schema,
+            pre_delete_table=pre_delete_table,
+        )
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "HologresVectorStore"
+
+    @property
+    def client(self) -> Any:
+        return self._storage
+
+    def add(
+        self,
+        nodes: List[BaseNode],
+        **add_kwargs: Any,
+    ) -> List[str]:
+        """Add nodes to hologres index.
+
+        Embedding data will be saved to `vector` column and text will be saved to `document` column.
+
+        Args:
+            nodes: List[BaseNode]: list of nodes with embeddings
+        """
+        embeddings = []
+        node_ids = []
+        schema_data_list = []
+        meta_data_list = []
+
+        for node in nodes:
+            text_embedding = node.get_embedding()
+            embeddings.append(text_embedding)
+            node_ids.append(node.node_id)
+            meta_data_list.append(node.metadata)
+            schema_data_list.append(
+                {"document": node.get_content(metadata_mode=MetadataMode.NONE)}
+            )
+
+        self._storage.upsert_vectors(
+            embeddings, node_ids, meta_data_list, schema_data_list
+        )
+        return node_ids
+
+    def query(
+        self,
+        query: VectorStoreQuery,
+        **kwargs: Any,
+    ) -> VectorStoreQueryResult:
+        """Query index for top k most similar nodes.
+
+        Args:
+            query_embedding (List[float]): query embedding
+            similarity_top_k (int): top k most similar nodes
+
+        """
+        query_embedding = cast(List[float], query.query_embedding)
+        top_k = query.similarity_top_k
+
+        query_results: List[dict[str, Any]] = self._storage.search(
+            query_embedding,
+            k=top_k,
+            select_columns=["document", "vector"],
+            metadata_filters=query.filters,
+        )
+
+        # if empty, then return an empty response
+        if len(query_results) == 0:
+            return VectorStoreQueryResult(similarities=[], ids=[])
+
+        nodes = []
+        similarities = []
+        ids = []
+
+        for result in query_results:
+            node = TextNode(
+                text=result["document"], id_=result["id"], embedding=result["vector"]
+            )
+            nodes.append(node)
+            ids.append(result["id"])
+            similarities.append(math.exp(-result["distance"]))
+
+        return VectorStoreQueryResult(nodes=nodes, similarities=similarities, ids=ids)
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        """
+        Delete nodes using with ref_doc_id.
+
+        Args:
+            ref_doc_id (str): The doc_id of the document to delete.
+
+        """
+        self._storage.delete_vectors(schema_data_filters={"id": ref_doc_id})
```

Comment:
Its not actually `id==ref_doc_id`, but meant to be a filtered delete on the metadata, where `metadata['doc_id'] == ref_doc_id`

Reply:
Yes. This is a bug. I have fixed in the latest commit and corrected the delete test case.